x-pack/plugins/osquery/server/routes/pack/create_pack_route.ts

```diff
@@ -174,7 +176,7 @@ export const createPackRoute = (router: IRouter, osqueryContext: OsqueryAppConte
 
       return response.ok({
         body: {
-          data: packSO,
+          data: { ...packSO.attributes, saved_object_id: packSO.id },
```

Comment:
Doesn't this still include the `packSO` type directly in the API response? We need to have API response types that are explicitly defined separately from the SO type (e.g. `PackSavedObjectLimited`).

I'm working on extending Dmitrii's [OpenAPI code generation](https://github.com/elastic/kibana/pull/157477) draft as a way to define API request and response types independently from the implementation. That would be a good way to ensure that the API types don't change accidentally when the internal SO types change in the future.